osquery/tables/system/windows/windows_crashes.cpp

```diff
@@ -0,0 +1,712 @@
+  registers << " rsp:0x" << std::hex << threadContext->Rsp;
+  registers << " r8:0x" << std::hex << threadContext->R8;
+  registers << " r9:0x" << std::hex << threadContext->R9;
+  registers << " r10:0x" << std::hex << threadContext->R10;
+  registers << " r11:0x" << std::hex << threadContext->R11;
+  registers << " r12:0x" << std::hex << threadContext->R12;
+  registers << " r13:0x" << std::hex << threadContext->R13;
+  registers << " r14:0x" << std::hex << threadContext->R14;
+  registers << " r15:0x" << std::hex << threadContext->R15;
+  registers << " rip:0x" << std::hex << threadContext->Rip;
+  registers << " segcs:0x" << std::hex << threadContext->SegCs;
+  registers << " segds:0x" << std::hex << threadContext->SegDs;
+  registers << " seges:0x" << std::hex << threadContext->SegEs;
+  registers << " segfs:0x" << std::hex << threadContext->SegFs;
+  registers << " seggs:0x" << std::hex << threadContext->SegGs;
+  registers << " segss:0x" << std::hex << threadContext->SegSs;
+  registers << " eflags:0x" << std::hex << threadContext->EFlags;
+  r["registers"] = registers.str();
+
+  return;
+}
+
+void processDumpMiscInfoStream(Row& r,
+                               MINIDUMP_DIRECTORY* dumpStreamDir,
+                               void* dumpStream,
+                               unsigned long dumpStreamSize) {
+  auto miscInfoStream = static_cast<MINIDUMP_MISC_INFO*>(dumpStream);
+
+  // Log PID, if it exists
+  if (miscInfoStream->Flags1 & MINIDUMP_MISC1_PROCESS_ID) {
+    r["pid"] = BIGINT(miscInfoStream->ProcessId);
+  }
+
+  // Log process times, if they exist
+  if (miscInfoStream->Flags1 & MINIDUMP_MISC1_PROCESS_TIMES) {
+    time_t procTimestamp = miscInfoStream->ProcessCreateTime;
+    struct tm gmt;
+    char timeBuff[64];
+    gmtime_s(&gmt, &procTimestamp);
+    strftime(timeBuff, sizeof(timeBuff), "%Y-%m-%d %H:%M:%S UTC", &gmt);
+    r["process_create_time"] = timeBuff;
+  }
+
+  return;
+}
+
+void processDumpSystemInfoStream(Row& r,
+                                 MINIDUMP_DIRECTORY* dumpStreamDir,
+                                 void* dumpStream,
+                                 unsigned long dumpStreamSize) {
+  auto systemInfoStream = static_cast<MINIDUMP_SYSTEM_INFO*>(dumpStream);
+
+  // Log system version information
+  r["major_version"] = INTEGER(systemInfoStream->MajorVersion);
+  r["minor_version"] = INTEGER(systemInfoStream->MinorVersion);
+  r["build_number"] = INTEGER(systemInfoStream->BuildNumber);
+
+  return;
+}
+
+void processDumpModuleListStream(Row& r,
+                                 MINIDUMP_DIRECTORY* dumpStreamDir,
+                                 void* dumpStream,
+                                 unsigned long dumpStreamSize,
+                                 void* dumpBase) {
+  auto moduleListStream = static_cast<MINIDUMP_MODULE_LIST*>(dumpStream);
+
+  // Log PE path
+  MINIDUMP_MODULE exeModule = moduleListStream->Modules[0];
+  auto exePath = reinterpret_cast<MINIDUMP_STRING*>(
```

Comment:
It seems it might be the case that we want `reinterpret_cast`s for a few of these binary-blob manipulations, however we really try to avoid making use of `reinterpret_cast`s at all costs, just an FYI. Is there any way to avoid needing a `reinterpret_cast` here? Maybe having a strong type on the function parameter? Not sure that'll solve it though.

Reply:
From what I've found, it looks like the `reinterpret_cast`s are inevitable when dealing with RVAs (e.g. line 56: https://reviews.llvm.org/differential/changeset/?ref=241973&whitespace=ignore-most) or values that should be treated as memory addresses (in our case, this is usually a target address to pull from the Minidump's stored memory). I was able to clean up a few of these though -- if you have any further suggestions, I'm open :)